### ISSUE TITLE
Failed flushing last frame when path does not found 

### ISSUE THREAD

**Issue by kazuki43zoo** [reactor/reactor-netty#103]
I've tried the `HttpServer` as follow:

```java
package com.example.netty;

import java.nio.file.FileSystems;
import java.util.function.Consumer;

import reactor.ipc.netty.NettyContext;
import reactor.ipc.netty.http.server.HttpServer;
import reactor.ipc.netty.http.server.HttpServerRoutes;

public class ReactorNettyApplication {

	public static void main(String... args) throws InterruptedException {
		NettyContext context = HttpServer.create(8080).newRouter(router()).block();
		Runtime.getRuntime().addShutdownHook(new Thread(context::dispose));
		context.channel().closeFuture().sync();
	}

	private static Consumer<? super HttpServerRoutes> router() {
		return routes -> {
			routes.directory("/", FileSystems.getDefault().getPath("src/main/webapp"));
		};
	}

}
```
```
$ curl -s -D - http://localhost:8080/foo
HTTP/1.1 404 Not Found
content-length: 0

```

A `EncoderException` is occurred on server log.

```
...
02:57:51.123 [reactor-http-nio-2] DEBUG reactor.ipc.netty.channel.ChannelOperations - [HttpServer] [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 - R:/0:0:0:0:0:0:0:1:61601] handler is being applied: reactor.ipc.netty.http.server.DefaultHttpServerRoutes@79e198e1
02:57:51.124 [reactor-http-nio-2] DEBUG reactor.ipc.netty.channel.ChannelOperationsHandler - [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 - R:/0:0:0:0:0:0:0:1:61601] Writing object DefaultFullHttpResponse(decodeResult: success, version: HTTP/1.1, content: EmptyByteBufBE)
HTTP/1.1 404 Not Found
content-length: 0
02:57:51.131 [reactor-http-nio-2] DEBUG reactor.ipc.netty.http.server.HttpServer - [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 - R:/0:0:0:0:0:0:0:1:61601] WRITE: 45B
         +-------------------------------------------------+
         |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f |
+--------+-------------------------------------------------+----------------+
|00000000| 48 54 54 50 2f 31 2e 31 20 34 30 34 20 4e 6f 74 |HTTP/1.1 404 Not|
|00000010| 20 46 6f 75 6e 64 0d 0a 63 6f 6e 74 65 6e 74 2d | Found..content-|
|00000020| 6c 65 6e 67 74 68 3a 20 30 0d 0a 0d 0a          |length: 0....   |
+--------+-------------------------------------------------+----------------+
02:57:51.131 [reactor-http-nio-2] DEBUG reactor.ipc.netty.http.server.HttpServer - [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 - R:/0:0:0:0:0:0:0:1:61601] FLUSH
02:57:51.134 [reactor-http-nio-2] DEBUG reactor.ipc.netty.http.server.HttpServerOperations - Last HTTP response frame
02:57:51.134 [reactor-http-nio-2] DEBUG reactor.ipc.netty.channel.ChannelOperationsHandler - [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 - R:/0:0:0:0:0:0:0:1:61601] Writing object EmptyLastHttpContent
02:57:51.134 [reactor-http-nio-2] DEBUG reactor.ipc.netty.http.server.HttpServer - [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 - R:/0:0:0:0:0:0:0:1:61601] FLUSH
02:57:51.135 [reactor-http-nio-2] DEBUG reactor.ipc.netty.http.server.HttpServer - [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 - R:/0:0:0:0:0:0:0:1:61601] USER_EVENT: [Handler Terminated]
02:57:51.136 [reactor-http-nio-2] DEBUG reactor.ipc.netty.channel.ChannelOperationsHandler - [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 - R:/0:0:0:0:0:0:0:1:61601] Disposing context reactor.ipc.netty.channel.ServerContextHandler@7adda6e8
02:57:51.138 [reactor-http-nio-2] ERROR reactor.ipc.netty.http.server.HttpServerOperations - Failed flushing last frame
io.netty.handler.codec.EncoderException: java.lang.IllegalStateException: unexpected message type: LastHttpContent$1
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:106)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:730)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:816)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:723)
	at reactor.ipc.netty.http.server.HttpServerHandler.write(HttpServerHandler.java:166)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:801)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:814)
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:794)
	at reactor.ipc.netty.channel.ChannelOperationsHandler.doWrite(ChannelOperationsHandler.java:272)
	at reactor.ipc.netty.channel.ChannelOperationsHandler.drain(ChannelOperationsHandler.java:401)
	at reactor.ipc.netty.channel.ChannelOperationsHandler.flush(ChannelOperationsHandler.java:179)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:776)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:802)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:814)
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:794)
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:831)
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1032)
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:296)
	at reactor.ipc.netty.http.server.HttpServerOperations.onOutboundComplete(HttpServerOperations.java:398)
	at reactor.ipc.netty.channel.ChannelOperations.onComplete(ChannelOperations.java:212)
	at reactor.core.publisher.Operators.complete(Operators.java:189)
	at reactor.ipc.netty.FutureMono$DeferredFutureMono.subscribe(FutureMono.java:125)
	at reactor.ipc.netty.channel.ChannelOperations.applyHandler(ChannelOperations.java:382)
	at reactor.ipc.netty.http.server.HttpServerOperations.onHandlerStart(HttpServerOperations.java:354)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:442)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: unexpected message type: LastHttpContent$1
	at io.netty.handler.codec.http.HttpObjectEncoder.encode(HttpObjectEncoder.java:99)
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:88)
	... 30 common frames omitted
02:57:51.140 [reactor-http-nio-2] DEBUG reactor.ipc.netty.http.server.HttpServer - [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 ! R:/0:0:0:0:0:0:0:1:61601] INACTIVE
02:57:51.140 [reactor-http-nio-2] DEBUG reactor.ipc.netty.http.server.HttpServer - [id: 0x3b03a252, L:/0:0:0:0:0:0:0:1:8080 ! R:/0:0:0:0:0:0:0:1:61601] UNREGISTERED
```

I think this behavior is bug. What do you think ?

### Affected version

* 0.6.3.RELEASE

**Comment by smaldini** [comment]
If you have a chance to try on 0.6.4.BUILD-SNAPSHOT

**Comment by smaldini** [comment]
Should be fixed with #91 - tried locally
